components/gitpod-db/go/personal_access_token.go

```diff
@@ -87,6 +87,30 @@ func CreatePersonalAccessToken(ctx context.Context, conn *gorm.DB, req PersonalA
 	return token, nil
 }
 
+func DeletePersonalAccessTokenForUser(ctx context.Context, conn *gorm.DB, tokenID uuid.UUID, userID uuid.UUID) (int64, error) {
+	if tokenID == uuid.Nil {
+		return 0, fmt.Errorf("Invalid or empty tokenID")
+	}
+
+	if userID == uuid.Nil {
+		return 0, fmt.Errorf("Invalid or empty userID")
+	}
+
+	db := conn.WithContext(ctx)
+
+	db = db.
+		Table((&PersonalAccessToken{}).TableName()).
+		Where("id = ?", tokenID).
+		Where("userId = ?", userID).
+		Where("deleted = ?", 0).
+		Update("deleted", 1)
+	if db.Error != nil {
+		return 0, fmt.Errorf("Failed to delete token: %v", db.Error)
+	}
```

Comment:
Should handle the case if the token doesn't exist, and return a `db.ErrorNotFound`. We should then check for that error in the API handler and return a NotFound to the user